### PR TITLE
pyproject: Use hatchling as build system

### PR DESCRIPTION
**Pull request by patrickelectric**
Well this works, but I would like the confirmation from the maturin team

https://github.com/PyO3/maturin/discussions/1714

### CODE REVIEW

**Comment by RaulTrombin** [comment]
Already merged required changes: https://github.com/pypa/hatch/pull/1018
Waiting a new release from hatch.

**Comment by RaulTrombin** [comment]
@patrickelectric 
https://github.com/pypa/hatch/releases/tag/hatchling-v1.19.0

**Comment by patrickelectric** [comment]
@RaulTrombin thanks!

File: .github/workflows/action.yml [hunk 1]
```diff
@@ -32,8 +32,9 @@ jobs:
       run: cargo build
     - name: Build Python module
       run: |
-        pip install --user maturin
-        pip install . --user
+        pip install --user maturin hatch
+        hatch run dev:build
+        hatch run dev:install
     - name: Check Python integration
       run: python examples/python/main.py
```
Comment (RaulTrombin) [inline .github/workflows/action.yml:39]:
```suggestion
      run: hatch run dev:examples/python/main.py
```

File: .github/workflows/action.yml [hunk 2]
```diff
@@ -32,8 +32,9 @@ jobs:
       run: cargo build
     - name: Build Python module
       run: |
-        pip install --user maturin
-        pip install . --user
+        pip install --user maturin hatch
+        hatch run dev:build
```
Comment (RaulTrombin) [inline .github/workflows/action.yml:36]:
```suggestion
```

**Review by RaulTrombin — COMMENTED** [review]
Install (using 'maturin develop') -> already compile and install alongside Python.

File: .github/workflows/action.yml (hunk 1 above)
Comment (patrickelectric) [reply .github/workflows/action.yml]: fixed

File: .github/workflows/action.yml (hunk 2 above)
Comment (patrickelectric) [reply .github/workflows/action.yml]: fixed

**Review by RaulTrombin — APPROVED** [review]

**Pull request merged**